# Week 9 - Estimation

This is a Jupyter notebook to explore the material in (Ross, 2017, Chp. 9). 



In [1]:
%matplotlib inline
# from now on we'll start each notebook with the library imports
# and special commands to keep these things in one place (which
# is good practice). The line above is jupyter command to get 
# matplotlib to plot inline (between cells)
# Next we import the libraries and give them short names
import numpy as np
import scipy.stats as stats
import matplotlib.pyplot as plt
from collections import Counter
from collections import defaultdict

## Exercise A

Complete questions 1 and 3  from (Ross, 2017, Sec. 9.2 Problems). The text is repeated below for convenience:

> 1. Consider a trial in which a jury must decide between hypothesis A that
the defendant is guilty and hypothesis B that he or she is innocent.
(a) In the framework of hypothesis testing and the U.S. legal system,
which of the hypotheses should be the null hypothesis?
(b) What do you think would be the appropriate significance level in
this situation?

> 3. Suppose a test of
>
>    $H_0 : \mu = 0$ against
>
>    $H_1 : \mu \neq 0$
>
>    resulted in rejection of H 0 at the 5 percent level of significance. Which of
the following statements is (are) accurate?
>
>    (a) The data proved that μ is significantly different from 0, meaning
that it is far away from 0.
>
>    (b) The data were significantly strong enough to conclude that μ is not
equal to 0.
>
>    (c) The probability that μ is equal to 0 is less than 0.05.
>
>    (d) The hypothesis that μ is equal to 0 was rejected by a procedure that
would have resulted in rejection only 5 percent of the time when μ
is equal to 0.

*complete your answers in Markdown*

### Exercise A
#### Question 1

Consider a trial in which a jury must decide between hypothesis A that the defendant is guilty and hypothesis B that he or she is innocent. 

(a) In the framework of hypothesis testing and the U.S. legal system, which of the hypotheses should be the null hypothesis? 

$H_0 : $ Hypothesis B - defendant that he or she is innocent

$H_1 : $ Hypothesis A - defendant is guilty

(b) What do you think would be the appropriate significance level in this situation?

Significance level should be small ($\alpha{}$ = 0.10, 0.05, and 0.01)

#### Question 3

Suppose a test of

$H_0$ : $\mu$ = $0$ against

$H_1$ : $\mu$ $\neq$ $0$

resulted in rejection of H 0 at the 5 percent level of significance. Which of
the following statements is (are) accurate?

(a) The data proved that μ is significantly different from 0, meaning
that it is far away from 0.

This statement is not accurate as it only proved that μ is not equal to 0

(b) The data were significantly strong enough to conclude that μ is not
equal to 0.

This statement is accurate as there was a 95% of level of significance for this case

(c) The probability that μ is equal to 0 is less than 0.05.

This statement is not accurate, as it is strictly 5% or less

(d) The hypothesis that μ is equal to 0 was rejected by a procedure that
would have resulted in rejection only 5 percent of the time when μ
is equal to 0.

This statement is not accurate as the procedure and probability is always the same for the outcome being not equal to or equal to 0.


### Exercise B

Complete question 1 from Problems for (Ross, 2017, Sec. 9.3) and question 1 from Problems for (Ross, 2017, Sec. 8.3.1). The text of each is repeated below for convenience:

#### Section 9.3(.0)
> 1. The device that an astronomer utilizes to measure distances results in
measurements that have a mean value equal to the actual distance of the
object being surveyed and a standard deviation of 0.5 light-years. Present
theory indicates that the actual distance from Earth to the asteroid Phyla
is 14.4 light-years. Test this hypothesis, at the 5 percent level of significance, 
if six independent measurements yielded the data
>
>    15.1, 14.8, 14.0, 15.2, 14.7, 14.5

#### Section 9.3.1

> 1. The weights of salmon grown at a commercial hatchery are normally dis-
tributed with a standard deviation of 1.2 pounds. The hatchery claims
that the mean weight of this year’s crop is at least 7.6 pounds. Suppose a
random sample of 16 fish yielded an average weight of 7.2 pounds. Is this
strong enough evidence to reject the hatchery’s claims at the
>
>    (a) 5 percent level of significance?
>
>    (b) 1 percent level of significance?
>
>    (c) What is the p value?


*complete your answers in Markdown using the code-block below for computation*

In [2]:
## supporting code for Exercise B

print("Exercise B")

"""
The device that an astronomer utilizes to measure distances results in measurements that have a mean 
value equal to the actual distance of the object being surveyed and a standard deviation of 0.5 light-years. 
Present theory indicates that the actual distance from Earth to the asteroid Phyla is 14.4 light-years. 
Test this hypothesis, at the 5 percent level of significance, if six independent measurements yielded the data

15.1, 14.8, 14.0, 15.2, 14.7, 14.5
"""

print("Section 9.3.0 - Question 1")
sd = 0.5
print(f"Standard Deviation = {sd}")
mu = 14.4
print(f"Mu = {mu}")
print("H0 (Null Hypothesis): mu = mu0 = 14.4")
print("H1: mu != mu0 = 14.4")
sl = 0.05
print(f"Significance Level = {sl}")
data = np.array([15.1, 14.8, 14.0, 15.2, 14.7, 14.5])
print(f"Data = {data}")
n = len(data)
print(f"Number of Data Values = {n}")
x_bar = np.sum(data)/n
print(f"X bar = {x_bar}")
standard_error = sd/(n**0.5)
print(f"Standard Error = {standard_error}")
t_test = (x_bar-mu)/standard_error
print(f"T Test = {t_test}")
print("Now, we need to find p value using t test at 0.05 significance level with degree of freedom (6 - 1) = 5")
dof = n-1
print(f"Degrees of freedom = {dof}")
p_val = stats.t.sf(t_test, dof)*2
print(f"P-value = {p_val}")
print(f"P-value = {round(p_val,3)}")
print("Here, the p value is greater than significance level (0.05) so we can't reject the null hypothesis")
print('\n')

"""
The weights of salmon grown at a commercial hatchery are normally distributed with a standard deviation of 1.2 pounds. 
The hatchery claims that the mean weight of this year’s crop is at least 7.6 pounds. 
Suppose a random sample of 16 fish yielded an average weight of 7.2 pounds. 
Is this strong enough evidence to reject the hatchery’s claims at the

(a) 5 percent level of significance?

(b) 1 percent level of significance?

(c) What is the p value?
"""
print("Section 9.3.1 - Question 1")
print("(a) 5 percent level of significance?")
sd = 1.2
print(f"SD = {sd}")
mu = 7.6
print(f"Mu = {mu}")
n = 16
print(f"Sample Size = {n}")
sl = 0.05
print(f"Significance Level = {sl}")
print("H0 (Null Hypothesis): mu = mu0 = 14.4")
print("H1: mu != mu0 = 14.4")
x_bar = 7.2
print(f"x_bar = {x_bar}")
standard_error = sd/(n**0.5)
print(f"Standard Error = {standard_error}")
t_test = (x_bar-mu)/standard_error
print(f"T Test = {t_test}")
t_test = t_test * (-1)
cdf = stats.norm.cdf(t_test)
print(f"cdf = {cdf}")
p_val = 1-cdf
print(f"P_value = {round(p_val,4)}")
print(f"Therefore, since p-value ({round(p_val,4)}) is greater than all considerable significance levels (alpha = 0.01,0.05), we fail to reject the null hypothesis. In other words, we do not have enough evidence to disprove the claims.")


Exercise B
Section 9.3.0 - Question 1
Standard Deviation = 0.5
Mu = 14.4
H0 (Null Hypothesis): mu = mu0 = 14.4
H1: mu != mu0 = 14.4
Significance Level = 0.05
Data = [15.1 14.8 14.  15.2 14.7 14.5]
Number of Data Values = 6
X bar = 14.716666666666667
Standard Error = 0.20412414523193154
T Test = 1.5513435037626782
Now, we need to find p value using t test at 0.05 significance level with degree of freedom (6 - 1) = 5
Degrees of freedom = 5
P-value = 0.1815179651046933
P-value = 0.182
Here, the p value is greater than significance level (0.05) so we can't reject the null hypothesis


Section 9.3.1 - Question 1
(a) 5 percent level of significance?
SD = 1.2
Mu = 7.6
Sample Size = 16
Significance Level = 0.05
H0 (Null Hypothesis): mu = mu0 = 14.4
H1: mu != mu0 = 14.4
x_bar = 7.2
Standard Error = 0.3
T Test = -1.3333333333333317
cdf = 0.9087887802741319
P_value = 0.0912
Therefore, since p-value (0.0912) is greater than all considerable significance levels (alpha = 0.01,0.05), we fail to rejec

## Exercise C

Complete questions 3 and 18 from (Ross, 2017, Sec. 9.4 Problems). The text is repeated below for convenience:

> 3. To test the hypothesis that a normal population has mean 100, a random sample of size 10 is chosen. If the sample mean is 110, will you reject the
null hypothesis if the following is known?
>
>    (a) The population standard deviation is known to equal 15.
>
>    (b) The population standard deviation is unknown, and the sample standard deviation is 15.
>
>    Use the 5 percent level of significance.

>18. In 2001, entering students at a certain university had an average score of
542 on the verbal part of the SAT. A random sample of the scores of 20 students in the 2003 class resulted in these scores:
>
>    542, 490, 582, 511, 515, 564, 500, 602, 488, 512, 518,
522, 505, 569, 575, 515, 520, 528, 533, 515
>
>    Do the given data prove that the average score has decreased to below 542? Use the 5 percent level of significance.

For 18., also calculate the p-value.

*complete your answers in Markdown using the code-block below for computation*

In [3]:
## supporting code for exercise C

print("Exercise C")

"""
To test the hypothesis that a normal population has mean 100, a random sample of size 10 is chosen. 
If the sample mean is 110, will you reject the null hypothesis if the following is known?

(a) The population standard deviation is known to equal 15.

(b) The population standard deviation is unknown, and the sample standard deviation is 15.

Use the 5 percent level of significance.
"""

print("Question 3")
print("(a) The population standard deviation is known to equal 15.")
mu = 100
print(f"Mu0 = {mu}")
print("H0 (Null Hypothesis): mu = mu0 = 100")
print("H1: mu != mu0 = 100")
n = 10
print(f"Sample Size = {n}")
x_bar = 110
print(f"Sample Mean (x bar) = {x_bar}")
sl = 0.05
print(f"Significance Level = {sl}")
sd = 15
print(f"SD = {sd}")
standard_error = sd/(n**0.5)
print(f"Standard Error = {standard_error}")
t_test = (x_bar-mu)/standard_error
print(f"T Test = {t_test}")
print("Now, we need to find p value using t test at 0.05 significance level with degree of freedom (10 - 1) = 9")
dof = n
print(f"Degrees of freedom = {dof}")
p_val = stats.t.sf(t_test, dof)*2
print(f"P-value = {p_val}")
print(f"P-value = {round(p_val,3)}")
print("Here, the p value is greater than significance level (0.05) so we can't reject the null hypothesis")
print('\n')

print("(b) The population standard deviation is unknown, and the sample standard deviation is 15.")
mu = 100
print(f"Mu0 = {mu}")
print("H0 (Null Hypothesis): mu = mu0 = 100")
print("H1: mu != mu0 = 100")
n = 10
print(f"Sample Size = {n}")
x_bar = 110
print(f"Sample Mean (x bar) = {x_bar}")
sl = 0.05
print(f"Significance Level = {sl}")
s_sd = 15
print(f"SD = {sd}")
standard_error = s_sd/(n**0.5)
print(f"Standard Error = {standard_error}")
t_test = (x_bar-mu)/standard_error
print(f"T Test = {t_test}")
print("Now, we need to find p value using t test at 0.05 significance level with degree of freedom (10 - 1) = 9")
dof = n-1
print(f"Degrees of freedom = {dof}")
p_val = stats.t.sf(t_test, dof)*2
print(f"P-value = {p_val}")
print(f"P-value = {round(p_val,3)}")
print("Here, the p value is greater than significance level (0.05) so we can't reject the null hypothesis")
print('\n')



Exercise C
Question 3
(a) The population standard deviation is known to equal 15.
Mu0 = 100
H0 (Null Hypothesis): mu = mu0 = 100
H1: mu != mu0 = 100
Sample Size = 10
Sample Mean (x bar) = 110
Significance Level = 0.05
SD = 15
Standard Error = 4.743416490252569
T Test = 2.1081851067789197
Now, we need to find p value using t test at 0.05 significance level with degree of freedom (10 - 1) = 9
Degrees of freedom = 10
P-value = 0.061228795670166467
P-value = 0.061
Here, the p value is greater than significance level (0.05) so we can't reject the null hypothesis


(b) The population standard deviation is unknown, and the sample standard deviation is 15.
Mu0 = 100
H0 (Null Hypothesis): mu = mu0 = 100
H1: mu != mu0 = 100
Sample Size = 10
Sample Mean (x bar) = 110
Significance Level = 0.05
SD = 15
Standard Error = 4.743416490252569
T Test = 2.1081851067789197
Now, we need to find p value using t test at 0.05 significance level with degree of freedom (10 - 1) = 9
Degrees of freedom = 9
P-value 

## Exercise D

Complete questions 1. and 14. from problems for (Ross, 2017, Sec. 9.5). The text is repeated below for convenience:


> 1. A standard drug is known to be effective in 72 percent of cases in which it is used to treat a certain infection. A new drug has been developed, and testing has found it to be effective in 42 cases out of 50. Is this strong enough evidence to prove that the new drug is more effective than the old one? Find the relevant p value.

Would you reject this at (a) the 5% level and (b) at the 1% level?


> 14. A statistics student wants to test the hypothesis that a certain coin is
equally likely to land on either heads or tails when it is flipped. The student flips the coin 200 times, obtaining 116 heads and 84 tails.
>
>    (a) For the 5 percent level of significance, what conclusion should be drawn?
>
>    (b) What are the null and the alternative hypotheses?
>
>    (c) What is the p value?


*complete your answers in Markdown using the code-block below for computation*

In [4]:
## supporting code for Exercise D

print("Exercise D")

"""
A standard drug is known to be effective in 72 percent 
of cases in which it is used to treat a certain infection. 
A new drug has been developed, and testing has found it 
to be effective in 42 cases out of 50. Is this strong 
enough evidence to prove that the new drug is more effective 
than the old one? Find the relevant p value.
"""
print("Question 1")
print("H0: p <= 0.72")
print("H1: p > 0.72")
print("In hope to eject hypothesis. The p-value of this test is equal to:")
n = 50
print(f"n = {n}")
X = 42
print(f"X = {X}")
p = X/n
print(f"p = {p}")
print(f"Binomial({n},{p})")
print(f"But given mean p = 72%, Binomial({n},0.72)")
p_val = 0.0365
print(f"p value = {p_val}")
print("Where we have used that X|H0 ~ Binom(50,0.72)")
print("So if consider sl = 0.1,0.05, we would reject null hypothesis")
print("and conclude new method is more efficient")
print("However, if alpha = 0.01, we would not be able to make that conclusion")
print('\n')

"""
A statistics student wants to test the hypothesis 
that a certain coin is equally likely to land on 
either heads or tails when it is flipped. 
The student flips the coin 200 times, obtaining 116 
heads and 84 tails.

(a) For the 5 percent level of significance, 
what conclusion should be drawn?

(b) What are the null and the alternative hypotheses?

(c) What is the p value?
"""

print("Question 14")
n = 200
print(f"n = {n}")
x = 54
print(f"x = {x}")
p0 = 0.5
print(f"p0 = {p0}")
mu = 100
np0 = 100
print(f"mu = {mu}, np0 = {np0}")
sd = (np0*(1-p0))**0.5
print(f"sd  {sd}")
sl = 0.05
print(f"Significance Level = {sl}")
print("H0: p = 0.5")
print("H1: p != 0.5")
test_stat = 116
print(f"Test Statistics (X) = {test_stat} (successes were taken to be heads)")
print("p value = 2[Min(P[X <= 116], P{X>=116})]")
print("Now, P[X<116] = P{X <= 116.5} apprx = P{Z<=(116.5-100)/7.0711}")
print("P[Z <= 2.3334] = 0.9901, and P{X >= 116} approx = 0.0141")
p_val = 0.0282
print(f"Hence, p value = {p_val}")
print("Therefore, p val is less than 0.05 sl")
print("Hence, H0 is rejected at the 5% level of significance ")
print("That is, the data are consisent with the alternative hupothesis that the proportion of heads (tails) is not equal to 50%")



Exercise D
Question 1
H0: p <= 0.72
H1: p > 0.72
In hope to eject hypothesis. The p-value of this test is equal to:
n = 50
X = 42
p = 0.84
Binomial(50,0.84)
But given mean p = 72%, Binomial(50,0.72)
p value = 0.0365
Where we have used that X|H0 ~ Binom(50,0.72)
So if consider sl = 0.1,0.05, we would reject null hypothesis
and conclude new method is more efficient
However, if alpha = 0.01, we would not be able to make that conclusion


Question 14
n = 200
x = 54
p0 = 0.5
mu = 100, np0 = 100
sd  7.0710678118654755
Significance Level = 0.05
H0: p = 0.5
H1: p != 0.5
Test Statistics (X) = 116 (successes were taken to be heads)
p value = 2[Min(P[X <= 116], P{X>=116})]
Now, P[X<116] = P{X <= 116.5} apprx = P{Z<=(116.5-100)/7.0711}
P[Z <= 2.3334] = 0.9901, and P{X >= 116} approx = 0.0141
Hence, p value = 0.0282
Therefore, p val is less than 0.05 sl
Hence, H0 is rejected at the 5% level of significance 
That is, the data are consisent with the alternative hupothesis that the proportion of heads

# LECTURE SLIDES TOPICS

## Tests concerning mean of normal population with known variance

https://www.youtube.com/watch?v=2GU_R7G5m-8&ab_channel=zedstatistics


### One Tailed Example

Question:

The manager of a department store is thinking about establishing a new billing system for the store’s credit customers. 
She determines that the new system will be cost effective only if the mean monthly account is greater than £70. 
A random sample of 200 monthly accounts is drawn for which the sample mean account is £74. 
The manager knows that the accounts are normally distributed with a standard deviation of £30. 
Is there enough evidence at the 5 percent level of significance to conclude that the new system will be cost effective?

Answer:

##### STEP 1: STATE NULL AND ALTERNATE HYPOTHESES

Tip: Always put what you are testing for in the alternate hypothesis ($H_1$)

$H_0: $ $\mu{}$ = 70 (some people put less than)

$H_1: $ $\mu{}$ > 70

##### STEP 2: CALCULATE TEST STATISTIC

$\bar{X} = 74$

$\sigma = 30$

$n = 200$

Z = $\frac{\bar{X} - \mu{}}{\frac{\sigma{}}{\sqrt{n}}}$

Z = $\frac{74 - 70}{\frac{30}{\sqrt{200}}}$

Z = 1.8856 = 1.886 (3 s.f.)

##### STEP 3: CONSIDER DECISION RULE

Z = $\frac{\bar{X} - \mu{}}{\frac{\sigma{}}{\sqrt{n}}}$ ~ $N(0,1)$

5 percent significance level (0.05)

Z-crit = 1.645 (= NORM.S.INV(0.95))

Therefore, we reject of Z > 1.645

##### STEP 3a: CALCULATE p-VALUE

The p-value is the probability of getting a sample as extreme as ours, given the null hypothesis is TRUE

p-value = 1 - NORM.S.DIST(1.886,TRUE) = 0.030

TRUE = CDF, FALSE = PMF

##### STEP 4: STATE REJECTION DECISION

Reject $H_0$ at 5% level of significance as Z > 1.645 (Z = 1.8856)

OR

Reject $H_0$ at 5% level of significance as p < 0.05 (p = 0.030)

##### STEP 5: CONCLUSION

There is enough evidence at the 5% level of significance to suggest that the mean monthly account is greater than £70


### Two Tailed Example

Question:

A new toll road is being built and financed on the expectation that 8,500 cars will use it per day. In the first 30 days of its operation, a daily average of 8,120 cars were found to have used the toll road. Using the 1% level of significance, test whether the expectation was incorrect. (Assume that the distribution of daily road users is normally distributed with a standard devation of 950)

Answer:

##### STEP 1: STATE NULL AND ALTERNATE HYPOTHESES

$H_0:$ $\mu{}$ = 8500

$H_1:$ $\mu{}$ $\neq$ 8500

##### STEP 2: CALCULATE TEST STATISTIC

$\bar{X} = 8120$

$\sigma = 950$

$n = 30$

Z = $\frac{\bar{X} - \mu{}}{\frac{\sigma{}}{\sqrt{n}}}$

Z = $\frac{8120 - 8500}{\frac{950}{\sqrt{30}}}$

Z = -2.191 = -2.19 (3 s.f.)

##### STEP 3: CONSIDER DECISION RULE

Z = $\frac{\bar{X} - \mu{}}{\frac{\sigma{}}{\sqrt{n}}}$ ~ $N(0,1)$

1 percent significance level (0.01)

Because two-tailed: each tail gets 0.005

Z-crit = 2.576 (= NORM.S.INV(0.995) - this because we just want one tail value above and we will know the other one)

Therefore, we reject of Z > 2.576 or Z < -2.576

##### STEP 3a: CALCULATE p-VALUE

The p-value is the probability of getting a sample as extreme as ours, given the null hypothesis is TRUE

p-value = NORM.S.DIST(-2.191,TRUE) = 0.0142

No '1 - NORM' becaus it is on left handside, one tailed example had Z on right hand side

Because two-tailed test, values more extreme, therefore:

p-value = NORM.S.DIST(-2.191,TRUE) * 2 = 0.0285

TRUE = CDF, FALSE = PMF

##### STEP 4: STATE REJECTION DECISION

Do not reject $H_0$ at 1% level of significance as -2.576 < Z < 2.576 (Z = 2.191)

OR

Do not reject $H_0$ at 1% level of significance as p > 0.01 (p = 0.0285)

##### STEP 5: CONCLUSION

There is NOT enough evidence at the 1% level of significance to suggest that the daily average of cars using the road is DIFFERENT from 8,500



## The t-test for mean of normal population with unknown variance

https://www.youtube.com/watch?v=Z8E3hi8INZk&ab_channel=zedstatistics


Question:

An online fashion store called Showdonkey advertises that its average delivery time is less than six hours for local deliveries. A random sample of the amount of time taken to deliver packages to an address in Stanmore produced the following delivery times (rounded to the nearest hour): 

7 3 4 6 10 5 6 4 3 8 

Is there sufficient evidence to support Showdonkey's advertisement, at the 5% level of significance?

Answer:

##### STEP 1: STATE NULL AND ALTERNATE HYPOTHESES

$H_0: \mu{} = 6 $

$H_1: \mu{}  < 6$

##### STEP 2: CALCULATE TEST STATISTIC

7 3 4 6 10 5 6 4 3 8

$\bar{X} = 5.6$

Sample Variance = $S^2 = \frac{sum(X - \bar{X})^2}{n-1}$

Sample SD (S) = 2.27 

t = $\frac{\bar{X}-\mu{}}{\frac{S}{\sqrt{n}}}$

t = $\frac{5.6-6}{\frac{2.27}{\sqrt{10}}}$

t stat (t-dist), not Z, because don't know the population SD is

t = -0.557

##### STEP 3: CONSIDER DECISION RULE

t = $\frac{\bar{X}-\mu{}}{\frac{S}{\sqrt{n}}}$ ~ $t_{DF}$

Degree of Freedom ($t_{DF}$) = n - 1 = 10 - 1 = 9

t-crit = -1.833 (=T.INV(0.05,9))

Reject if t < -1.833

##### STEP 3a: CALCULATE p-VALUE

The p-value is the probability of getting a sample as extreme as ours, given the null hypothesis is TRUE

p-value = T.DIST(-0.557,9,TRUE) = 0.296

##### STEP 4: STATE REJECTION DECISION

Do not reject $H_0$ at 5% level of significance as t > -1.833 (t = -0.557)

OR

Do not reject $H_0$ at 5% level of significance as p > 0.05 (p = 0.296)

NOTE: You can check if you have correct conclusion if t stat and p value agree, else, something has been done wrong in calculation

##### STEP 5: CONCLUSION

While the sample mean ws 5.6 hours, there is NOT enough evidence at the 5% level of significance to infer that the population average time for delivery is less than 6 hours


## Hypothesis tests for population proportions

https://www.youtube.com/watch?v=bh_5Yr3YjO0&ab_channel=zedstatistics


Question:

Gigi's Pizza regularly surveys its customers. On the basis of these surveys, the management of the chain claims that at least 75% of its customers rate the food as excellent. A consumer testing service wants to examine the claim by asking 460 customers to rate the food. Seventy one per cent rated the food as excellent. Is there enough evidence at the 

(a) 5% and 

(b) 1% level of significance to reject management's claim?

Answer:

##### STEP 1: STATE NULL AND ALTERNATE HYPOTHESES

$H_0: \mu{} = 0.75$

$H_1: \mu{}  < 0.75$

##### STEP 2: CALCULATE TEST STATISTIC

$\hat{p}$ = 0.71

$p$ = 0.75

$n$ = 460

In this case we have a binomial distribution

As np > 5 and n(1-p) > 5, C.L.T (Central Limit Theorem) applies

Therefore, test statistic will be approximately normally distributed

$\frac{\hat{p} - p}{\sqrt{\frac{p(1-p)}{n}}}$ $approx =$ $N(0,1)$

Z = $\frac{\hat{p} - p}{\sqrt{\frac{p(1-p)}{n}}}$

Z = $\frac{0.71 - 0.75}{\sqrt{\frac{(0.75)(0.25)}{460}}}$

Z = -1.981

##### STEP 3: CONSIDER DECISION RULE 

Note: only one tailed hypotheses, if two tailed, split significance level

5% level of significance:

0.05 level of significance

Z-crit = -1.645 (=NORM.S.INV(0.05))

Reject if Z < -1.645

1% level of significance:

0.01 level of significance

Z-crit = -2.33 (=NORM.S.INV(0.01))

Reject if Z < -2.33

##### STEP 3a: CALCULATE p-VALUE

The p-value is the probability of getting a sample as extreme as ours, given the null hypothesis is TRUE

p-value = NORM.S.DIST(-1.981,TRUE) = 0.0238

##### STEP 4: STATE REJECTION DECISION

5% level of significance:

Reject $H_0$ at 5% level of significance as Z < -1.645 (Z = -1.981)

OR

Reject $H_0$ at 5% level of significance as p < 0.05 (p = 0.0238)

1% level of significance:

Do not reject $H_0$ at 1% level of significance as Z > -2.33 (Z = -1.981)

OR

Do not reject $H_0$ at 1% level of significance as p > 0.01 (p = 0.0238)


##### STEP 5: CONCLUSION

There is enough evidence at the 5% level of significance  (but not the 1% level of significance) to suggest that the proportion of customers rating Gigi's food as "excellent" is less than 75%.


# QUIZ 9+10

## QUESTION 1 - PART A

A scientist works with a colony of many thousands of mice. The average weight of all the mice is known to be 19.5 grammes with SD 3.4 grammes. She asks her first lab assistant to select 22 mice for an experiment, but is concerned that the selection of the mice may not be truly random and is instead is biased in some way. The selected mice may have been those that were slowest in avoiding the assistant, and this might influence the outcome of the experiment. The weights of the mice selected by the first assistant are as follows:

13.7,22.1,17.0,12.4,16.4,7.8,20.4,21.1,19.7,17.5,18.4,

25.2,15.1,23.0,18.3,23.9,14.8,20.6,17.2,16.0,18.4,15.5

She decides to perform a hypothesis test at 10% significance level to see if the weights recorded indicated that the sample was not consistent with a random sample from the colony. In particular, she wants to test the null hypothesis that the sample distributions mean is the same as the mean of the colony. Consider the most appropriate test:

### What is the value the test statistic? (give your answer to 3 decimal places)

In [5]:
data = np.array([13.7,22.1,17.0,12.4,16.4,7.8,20.4,21.1,19.7,17.5,18.4,25.2,15.1,23.0,18.3,23.9,14.8,20.6,17.2,16.0,18.4,15.5])
x_bar = np.sum(data)/len(data)
print(f"x_bar = {x_bar}")
mu = 19.5
print(f"mu = {mu}")
sd = 3.4
print(f"sd = {sd}")
n = 22
print(f"n = {n}")
test_stat = (x_bar-mu)/(sd/(n**0.5))
print(f"test_stat = {test_stat} = {round(test_stat,3)}")


x_bar = 17.931818181818183
mu = 19.5
sd = 3.4
n = 22
test_stat = -2.163360210078986 = -2.163


### What is the value of the (z-percentile) threshold use for the significance level test? (give your answer to 3 decimal places)

In [6]:
sl = 0.1
print(f"sl = {sl}")
print(f"Because two tailed, each tail gets: {sl/2}")
Z = 1.644853627
print(f"Z = {Z} = {round(Z,3)}")


sl = 0.1
Because two tailed, each tail gets: 0.05
Z = 1.644853627 = 1.645


### What is the p-value of the test statistic? (give your answer to 3 significant figures)

In [7]:
pval = 0.030513485
print(f"{round(pval, 6)}")

0.030513


## QUESTION 1 - PART B

The scientist has two lab assistants and subsequently decides that so long as the second assistant doesn't appear to collect mice differently from the first the experiments can go ahead. She asks the second assistant to collect $12$ mice and decides to test whether there is a difference between the two assistants sampling distributions. The weights of the mice collected by the second assistant are as follows:

21.0,21.9,21.0,22.3,19.0,12.9,13.7,21.3,21.2,18.7,16.4,22.8

The scientist assumes that the variance of the two sampling approaches are unknown, but equal, and conducts the appropriate hypothesis test again at a 10% level of significance. Consider the appropriate hypothesis test:

In [8]:
data1 = np.array([13.7,22.1,17.0,12.4,16.4,7.8,20.4,21.1,19.7,17.5,18.4,25.2,15.1,23.0,18.3,23.9,14.8,20.6,17.2,16.0,18.4,15.5])
n = len(data1)
print(f"n = {n}")
x = np.sum(data1)/n
print(f"x = {x}")
data2 = np.array([21.0,21.9,21.0,22.3,19.0,12.9,13.7,21.3,21.2,18.7,16.4,22.8])
m = len(data2)
print(f"m = {m}")
y = np.sum(data2)/m
print(f"y = {y}")
sd1 = 4.0298158251235
print(f"sd1 = {sd1}")
sd2 = 3.337800037586
print(f"sd2 = {sd2}")
Sx2 = sd1**2
print(f"Sx2 = {Sx2}")
Sy2 = sd2**2
print(f"Sy2 = {Sy2}")
sd3 = (sd1+sd2)/2
print(f"sd3 = {sd3}")
Sp2 = sd3**2
print(f"Sp3 = {Sp2}")
sig_lev = 0.10
print(f"Significance Level = {sig_lev}")
alpha2 = sig_lev/2
print(f"Alpha/2 = {alpha2}")
dof = (n+m)-2
print(f"dof = {dof}")
dof_alpha2 = 1.694
print(f"t32,alpha/2 = {dof_alpha2}")
print("H0: muX = muY")
print("H1: muX != muY")
test_stat = (x-y)/((Sp2*((1/n)+(1/m)))**0.5)
print(f"test_stat = {test_stat}")
print(f"Conclusion, since |T| = {test_stat*-1} < {dof_alpha2}, we cannot reject H0")
print(f"p_value = {0.28372}")


n = 22
x = 17.931818181818183
m = 12
y = 19.349999999999998
sd1 = 4.0298158251235
sd2 = 3.337800037586
Sx2 = 16.2394155844158
Sy2 = 11.140909090909105
sd3 = 3.6838079313547505
Sp3 = 13.570440875112165
Significance Level = 0.1
Alpha/2 = 0.05
dof = 32
t32,alpha/2 = 1.694
H0: muX = muY
H1: muX != muY
test_stat = -1.072747323165922
Conclusion, since |T| = 1.072747323165922 < 1.694, we cannot reject H0
p_value = 0.28372


## QUESTION 2

An experimenter wishes to investigate whether average performance on a particular mobile game is degraded if a particular cognitively fatiguing task is carried out in advance. To do so, he brings in participants on two occasions 1 week apart, and asks them to play the game, on one occasion asking them to complete the fatiguing task first. To control for recent prior experience with the game, he randomises whether they take the fatiguing task before their first or second session.

He records the time it takes for each individual to complete a specific level in the game, with stronger players completing the level more quickly. The table shows the times recorded by each player during each session, and which session included the fatiguing task. He wishes to test whether times are longer after the fatiguing task as this would represent a significant finding.

The experimenter performs a paired sample t-test at 5% significance leve, and calculates the corresponding t-distributed test statistic:


In [9]:
nonf_task = np.array([197.0,100.3,75.5,122.6,100.7,88.9,85.3,99.6,101.9,159.3,118.2,126.8,145.1,150.1,97.2,77.6,96.8])
print(f"Non-Fatigue Task = {nonf_task}")
n = len(nonf_task)
print(f"Number of Non-Fatigue Data Values = {n}")
f_task = np.array([223.1,129.3,82.0,139.2,129.8,109.5,80.8,95.2,94.3,169.5,106.6,130.8,168.0,136.1,117.4,76.5,80.9])
print(f"Fatigue Task = {f_task}")
m = len(f_task)
print(f"Number of Fatigue Data Values = {m}")
diff = nonf_task - f_task
print(f"diff = {diff}")
d_bar = np.mean(diff)
print(f"d_bar = {d_bar}")
S_d = 15.633027349738
print(f"S_d = {S_d}")


Non-Fatigue Task = [197.  100.3  75.5 122.6 100.7  88.9  85.3  99.6 101.9 159.3 118.2 126.8
 145.1 150.1  97.2  77.6  96.8]
Number of Non-Fatigue Data Values = 17
Fatigue Task = [223.1 129.3  82.  139.2 129.8 109.5  80.8  95.2  94.3 169.5 106.6 130.8
 168.  136.1 117.4  76.5  80.9]
Number of Fatigue Data Values = 17
diff = [-26.1 -29.   -6.5 -16.6 -29.1 -20.6   4.5   4.4   7.6 -10.2  11.6  -4.
 -22.9  14.  -20.2   1.1  15.9]
d_bar = -7.41764705882353
S_d = 15.633027349738


### How many degrees of freedom does the test statistic have? (give the precise answer)

In [10]:
dof = n-1
print(f"dof = {dof}")

dof = 16


### What is the value of the test statistic? (give the answer to 3 decimal places)

In [11]:
test_stat = ((n**0.5)*d_bar)/S_d
print(f"test_stat = {test_stat} = {round(test_stat,3)}")


test_stat = -1.956354430454838 = -1.956


In [12]:
print("t_n-1,alpha/2 = t_16,0.05/2 = t_16,0.025")
thresholdvalue = 2.120
print(f"thresholdvalue = {thresholdvalue}")

t_n-1,alpha/2 = t_16,0.05/2 = t_16,0.025
thresholdvalue = 2.12


## QUESTION 3 - PART A

During the COVID pandemic, a researcher is interested in the social influences on the wearing of a mask. She is first interested in whether the sight of others wearing (or not wearing) masks affects the observer's choice to wear a mask. To investigate this, she records data at a shop in her local area.

When queuing outside to enter the shop, a customer must wait for another to leave the shop before entering. When they do, the researcher records whether the leaving customer is wearing a mask and whether the person entering then chooses to put on a mask or not.  

Table from quiz 9+10

Her recorded the data is shown in the Table. For example, she observed 212 customers put on a mask, after seeing someone leave who was not wearing one. She constructs the appropriate hypothesis test to test whether this data showed a difference of any kind between the whether or not the departing customer wore a mask or not.

In [13]:
prev_without_mask = 212+38
print(f"prev_without_mask = {prev_without_mask}")
X = 212
print(f"X (puts on mask when sees no mask) = {X}")
p1 = X/prev_without_mask
print(f"p1 = {p1}")
print('\n')

prev_with_mask = 1618+205
print(f"prev_with_mask = {prev_with_mask}")
Y = 1618
print(f"Y (puts on mask when sees mask) = {Y}")
p2 = Y/prev_with_mask
print(f"p2 = {p2}")
print('\n')

pooled_est = (X+Y)/(prev_without_mask+prev_with_mask)
print(f"pooled_est = {pooled_est}")
print('\n')

test_stat = (p1-p2)/((((1/prev_without_mask)+(1/prev_with_mask))*(pooled_est)*(1-pooled_est))**0.5)
print(f"test_stat = {test_stat} = {round(test_stat,3)}")
print('\n')


prev_without_mask = 250
X (puts on mask when sees no mask) = 212
p1 = 0.848


prev_with_mask = 1823
Y (puts on mask when sees mask) = 1618
p2 = 0.8875479978058146


pooled_est = 0.8827785817655571


test_stat = -1.8228830615365312 = -1.823




### What is the value of the appropriate test statistic for the this test (give your answer to 3 decimal places)

In [14]:
test_stat = (p1-p2)/((((1/prev_without_mask)+(1/prev_with_mask))*(pooled_est)*(1-pooled_est))**0.5)
print(f"test_stat = {test_stat} = {round(test_stat,3)}")


test_stat = -1.8228830615365312 = -1.823


### Indicate whether or not this should be rejected at the 5% significance level.

In [15]:
print("H0: mu1 >= mu2")
print("The number of people who put the mask on ")
print("H1: H0: mu1 < mu2")
Zalphaover2 = 1.960
print(f"{test_stat} < {Zalphaover2}")
print("Therefore, do not reject H0")


H0: mu1 >= mu2
The number of people who put the mask on 
H1: H0: mu1 < mu2
-1.8228830615365312 < 1.96
Therefore, do not reject H0


### What is the p-value of the test statistic? (give your answer to 3 significant figures) 

In [16]:
pval = 0.034227
print(f"p value = P(Z ≥ {test_stat}) = {pval}")


p value = P(Z ≥ -1.8228830615365312) = 0.034227


## QUESTION 3 - PART B

In a second study, our researcher reads a report that in a survey 81.5% of people reported that they felt it is important to wear masks when travelling on London buses. However, she fears that a recent news story has undermined people's willingness to wear masks. She wishes to test whether the proportion of people now wearing masks on London buses is lower than this number. She conducts a study where a random selection of bus commuters are observed and it is recorded whether or not they are wearing a mask. She observes 1554 people wearing masks appropriately, 420 not wearing masks, and 33 wearing masks inappropriately. She decides that those wearing masks inappropriately should be excluded as they do not inform either way about the hypothesis.


In [17]:
mask_appr = 1554
no_mask = 420 
mask_inappr = 33 

# NOTE THIS QUESTION IS ABOUT ONE POPULATION (PROPORTION) - MASK WEARERS - THEREFORE LECTURE 9 WILL BE USED


### What is the sample size?

In [18]:
sample_size = mask_appr + no_mask
print(f"Sample Size = {sample_size}")
print("H0: p <= p0")
print("H1: p > p0")


Sample Size = 1974
H0: p <= p0
H1: p > p0


### What is the associated test statistic? (give the precise value) 

In [19]:
n = sample_size
print(f"n = {n}")
X = 1554
print(f"X = {X}")
p_hat = X/n
print(f"P Hat = {p_hat}")
p = 0.85
print("We have a binomial distribution in this case")
np = n*p
print(f"np = {np} > 5")
secondtest = n*(1-p)
print(f"2ndtest = {secondtest} > 5")
print("Therefore, Central Limit Theorem (CLT) applies, therefore, test stat will be approx normally distributed ")
test_stat = (p_hat-p)/((p*(1-p))/n)**0.5
print(f"test_stat = {test_stat}")


n = 1974
X = 1554
P Hat = 0.7872340425531915
We have a binomial distribution in this case
np = 1677.8999999999999 > 5
2ndtest = 296.1 > 5
Therefore, Central Limit Theorem (CLT) applies, therefore, test stat will be approx normally distributed 
test_stat = -7.809849049611781


### What is the associated p-value? (give the answer to 3 significant figures) 

In [20]:
p_val = 0.000000000000002862825907486770000000
print(f"p value = {p_val}")

p value = 2.86282590748677e-15


### Can she conclude that the news story has affected peoples willingness to wear a mask? Give the most appropriate response

In [22]:
print(f"p value is extremely small, approx = 0, therefore, reject null hypothesis")
print(f"however, sample is not representative enough, also, inappropriate mask wearers were rejected, also other reasons could exist for the reduction of mask use")

p value is extremely small, approx = 0, therefore, reject null hypothesis
however, sample is not representative enough, also, inappropriate mask wearers were rejected, also other reasons could exist for the reduction of mask use
